src/InteractiveWindow/Editor/History.cs

```diff
@@ -159,11 +164,16 @@ private Entry Get(string pattern, int step)
 
         internal Entry GetNext(string pattern = null)
         {
+            //if you are at the last history entry or history navigation pointer is
+            //not initialized then return null as there is no next entry
+            if ((_current == Length - 1) || (_current == -1)) return null;
```

Comment:
Is ```Move``` really general purpose?  It has logic for stopping after a fixed number of steps which seems tied to a specific history browsing design.  Since it's a private method, I think it makes sense to remove the functionality we're not using and rebuild it later if our needs change.